e2e/e2e-script.sh

```diff
@@ -0,0 +1,164 @@
+if [ -z $(az aks list -g $RESOURCE_GROUP_NAME | jq '.[].name') ]; then
+    echo "Cluster doesnt exist, creating"
+    az aks create -g $RESOURCE_GROUP_NAME -n $CLUSTER_NAME --node-count 1 --generate-ssh-keys
+    az aks get-credentials -g $RESOURCE_GROUP_NAME -n $CLUSTER_NAME
+fi
+
+MC_RESOURCE_GROUP_NAME=$(az aks show -n $CLUSTER_NAME -g $RESOURCE_GROUP_NAME | jq -r '.nodeResourceGroup')
+VMSS_NAME=$(az vmss list -g $MC_RESOURCE_GROUP_NAME | jq -r '.[length -1].name')
+CLUSTER_ID=$(echo $VMSS_NAME | cut -d '-' -f3)
+
+# Retrieve the etc/kubernetes/azure.json file for cluster related info
+az vmss run-command invoke \
+            -n $VMSS_NAME \
+            -g $MC_RESOURCE_GROUP_NAME \
+            --command-id RunShellScript \
+            --instance-id 0 \
+            --scripts "cat /etc/kubernetes/azure.json" | jq -r '.value[].message' | awk '/{/{flag=1}/}/{print;flag=0}flag' \
+            > fields.json
+
+# Retrieve the keys and certificates
+
+# TODO 2: If TLS Bootstrapping is not enabled, the client.crt takes some time to be generated. The run-command throws
+#       an error saying that extension is still being applied. Need to introduce some delay before this piece of code is
+#       called and the file is ready to be read else the whole flow will break. 
+
+declare -a files=("apiserver.crt" "ca.crt" "client.key" "client.crt")
+for file in "${files[@]}"; do
+    content=$(az vmss run-command invoke \
+                -n $VMSS_NAME \
+                -g $MC_RESOURCE_GROUP_NAME \
+                --command-id RunShellScript \
+                --instance-id 0 \
+                --scripts "cat /etc/kubernetes/certs/$file" | \
+                jq -r '.value[].message' | \
+                awk '/stdout/{flag=1;next}/stderr/{flag=0}flag' | \
+                awk NF | base64 \
+            )
+    jq -r --arg key $file --arg value $content '. + { ($key) : $value }' < fields.json > dummy.json && mv dummy.json fields.json
+done
+
+# Add other relevant information needed by AgentBaker for bootstrapping later
+getAgentPoolProfileValues
+getFQDN
+getMSIResourceID
+
+addJsonToFile "mcRGName" $MC_RESOURCE_GROUP_NAME
+addJsonToFile "clusterID" $CLUSTER_ID
+addJsonToFile "subID" $SUBSCRIPTION_ID
+
+# Check if TLS Bootstrapping is enabled(no client.crt in that case, retrieve the tlsbootstrap token)
+tlsbootstrap=$(az vmss run-command invoke \
+                -n $VMSS_NAME \
+                -g $MC_RESOURCE_GROUP_NAME \
+                --command-id RunShellScript \
+                --instance-id 0 \
+                --scripts "cat /var/lib/kubelet/bootstrap-kubeconfig" | \
+                jq -r '.value[].message' | \
+                grep "token" | \
+                cut -f2 -d ":" | tr -d '"'
+            )
+
+if [[ -z ${tlsbootstrap} ]]; then
+    echo "TLS Bootstrap disabled"
+else
+    addJsonToFile "tlsbootstraptoken" $tlsbootstrap
+fi
+
+# Call AgentBaker to generate CustomData and cseCmd
+go test -v
```

Comment:
nit: call out the actual _test.go file for clarity? this is a critical step so the more clear the better

Reply:
why do we need this call at all?